homeassistant/components/onvif/camera.py

```diff
@@ -96,8 +96,8 @@ def name(self) -> str:
     def unique_id(self) -> str:
         """Return a unique ID."""
         if self.profile.index:
-            return f"{self.device.info.mac}_{self.profile.index}"
-        return self.device.info.mac
+            return f"{self.device.info.mac or self.device.info.serial_number}_{self.profile.index}"
```

Comment:
Is it the case that some cameras have mac address but not serial number and some cameras have serial number but not mac address?

Reply:
#35883 shows that some cameras don't return a MAC, but do return a SerialNumber.  I do not know of any cases where no serial number is returned, but wouldn't be surprised if there is one somewhere...

Reply:
The real reason for this statement is backwards compatibility for entities.  MAC was pulled and used as `unique_id` between my time working on this integration.  This way will only break entity mapping for devices that don't return a MAC, which are already currently completely broken, so not a huge deal breaker.